libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -510,3 +680,118 @@ void ContractLevelChecker::checkBaseABICompatibility(ContractDefinition const& _
 		);
 
 }
+
+void ContractLevelChecker::checkAmbiguousOverrides(ContractDefinition const& _contract) const
+{
+	vector<FunctionDefinition const*> contractFuncs = _contract.definedFunctions();
+
+	FunctionSet const& baseSet = getBaseFunctions(&_contract);
+	//auto const resolvedBases = resolveBaseContracts(_contract);
+
+	set<FunctionDefinition const*, LessFunction> processedFuncs;
+
+	for (FunctionDefinition const* function: baseSet)
```

Comment:
```
if (baseSet.empty()) return;
auto rangeStart = baseSet.begin();
for (auto it = std::next(rangeStart); it != baseSet.end(); ++it)
{
	if (LessFunction{}(*rangeStart, *it))
    		if (std::next(rangeStart) != it && !std::count_if(contractFuncs.cbegin(), contractFuncs.cend(), [](auto const* f) { return hasEqualNameAndParameters(*rangeStart, *f); }))
    		{
    			SecondarySourceLocation ssl;
			for (; rangeStart != it; ++rangeStart)
				ssl.append("Definition here: ", rangeStart->location());
			m_errorReporter.typeError(
				_contract.location(),
				ssl,
				"Functions of the same name " +
				function->name() +
				" and parameter types defined in two or more base contracts must be overridden in the derived contract."
			);
		}
}
```
or something like that?

Reply:
Right, taking advantage of the order is smarter here

Reply:
Yep, here I think it in fact makes sense to basically iterate over equal ranges. Not sure, if it might be easier to read, if we added a helper function for that or if we just do it in-place here...